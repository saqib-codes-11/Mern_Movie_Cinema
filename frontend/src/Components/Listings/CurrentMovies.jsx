'use strict';
import { useState, useEffect } from "react";
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Card from 'react-bootstrap/Card';
import Button from 'react-bootstrap/Button';
import axios from 'axios';
import { Link } from 'react-router-dom';
import Jumbotron from 'react-bootstrap/Jumbotron'
import './Movies.css';
import '../../CSS/Pages.css';

const CurrentMovies = () => {

    const [movieList, setMovieList] = useState([]);
    const [error, setError] = useState(null);
    const [loaded, setLoaded] = useState(false);
    const loadTime = 100;

    const getMovies = () => {
        axios.get('http://localhost:5000/released/true')
            .then((response) => {
                setLoaded(true);
                setMovieList(response.data);
            })
            .catch((error) => {
                setLoaded(true);
                setError(error);
            });
    }

    useEffect(() => {
        getMovies();
    }, []);

    const imageUpdater = (oldImage) => {
        return oldImage.replace('._V1_SX300', '')
    }

    if (!loaded) {
        return <p>Data is loading</p>
    }

    return (
        <div class="background">
            <Container>
                <Jumbotron className="bgBlur">
                    <h1 className='landing-text'>Current Movies </h1>
                    <h3 className="lead" >Come to our cinema and watch these films right now!</h3>
                    <div>
                        <Row>
                            {movieList.map((movie) => (
                                <Col lg={4} xs={12} md={6} className="p-2">
                                    <div className="card-wrapper">
                                        <div className="content">
                                            <div className="face-front z-depth-2">
                                                <img className="movieImage" src={imageUpdater(movie.imageURL)} alt={movie.title} />
                                            </div>
                                            <div className="face-back z-depth-2 colorScheme">
                                                <Card.Body >
                                                    <Card.Title>{movie.title}</Card.Title>
                                                    <Card.Title>Directed by {movie.director}</Card.Title>
                                                    <br />
                                                    <Card.Subtitle>Released: {movie.year}</Card.Subtitle>
                                                    <Link to={`/released/${movie._id}`}>
                                                        <Button className="alignBottom" value={movie._id} variant="dark">View more...</Button>
                                                    </Link>
                                                </Card.Body>
                                            </div>
                                        </div>
                                    </div>
                                </Col>
                            ))}
                        </Row>
                    </div>
                </Jumbotron>
            </Container>
        </div >
    )
}
export default CurrentMovies;