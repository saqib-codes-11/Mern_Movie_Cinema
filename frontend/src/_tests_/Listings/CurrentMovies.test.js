import CurrentMovies from '../../Components/Listings/CurrentMovies'
import { create } from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";
import React from 'react';

describe(`Current movies landing page testing`, () => {

    let creator;
    let testComponent;

    let movieObjs = [{
        title: "Fast and Furious 9",
        year: 2021,
        runTime: "143 mins",
        genre: "Action, Adventure, Crime",
        shortPlot: "Cipher enlists the help of Jakob, Dom's younger brother to take revenge on Dom and his team.",
        imageURL: "https://m.media-amazon.com/images/M/MV5BMjI0NmFkYzEtNzU2YS00NTg5LWIwYmMtNmQ1MTU0OGJjOTMxXkEyXkFqcGdeQXVyMjMxOTE0ODA@._V1_SX300.jpg",
        actors: [
            {
                name: "Vin Diesel",
                role: "Dominic Toretto",
                image: "https://www.themoviedb.org/t/p/w600_and_h900_bestv2/ac1dr6oK2ImdcJInKjIZw9sG6Jk.jpg"
            },
            {
                name: "Michelle Rodriguez",
                role: "Letty Ortiz",
                image: "https://www.themoviedb.org/t/p/w600_and_h900_bestv2/soXPKCMmOxT7oyfGsMUHB6YHLcC.jpg"
            },
            {
                name: "Tyrese Gibson",
                role: "Roman Pearce",
                image: "https://www.themoviedb.org/t/p/w600_and_h900_bestv2/jxoy4fbXNKFQtBdK73cLXEz3ufS.jpg"
            }
        ],
        director: "Justin Lin",
        dateTime: [
            {
                day: "Monday",
                timeOfMovie: [
                    {
                        time: "18:00"
                    },
                    {
                        time: "20:00"
                    },
                    {
                        time: "22:00"
                    }
                ]
            },
            {
                day: "Tuesday",
                timeOfMovie: [
                    {
                        time: "18:00"
                    },
                    {
                        time: "20:00"
                    },
                    {
                        time: "22:00"
                    }
                ]
            },
            {
                day: "Wednesday",
                timeOfMovie: [
                    {
                        time: "18:00"
                    },
                    {
                        time: "20:00"
                    },
                    {
                        time: "22:00"
                    }
                ]
            },
            {
                day: "Thursday",
                timeOfMovie: [
                    {
                        time: "18:00"
                    },
                    {
                        time: "20:00"
                    },
                    {
                        time: "22:00"
                    }
                ]
            },
            {
                day: "Friday",
                timeOfMovie: [
                    {
                        time: "18:00"
                    },
                    {
                        time: "20:00"
                    },
                    {
                        time: "22:00"
                    }
                ]
            }
        ],
        released: true,
        classification: "12A"
    }]

    beforeEach(() => {
        const TestInstance = create(<CurrentMovies />)
        testComponent = TestInstance.root;
    })

    it(`should render data is loading`, () => {
        const p = testComponent.findByType('p');
        expect(p.children[0]).toEqual('Data is loading');
    })

    it(`should render forum landing page`, async () => {
        const realUseState = React.useState

        jest.spyOn(React, 'useState')
            .mockImplementationOnce(() => realUseState(movieObjs))
            .mockImplementationOnce(() => realUseState(null))
            .mockImplementationOnce(() => realUseState(true))
        creator = create(<BrowserRouter><CurrentMovies /></BrowserRouter >)
        testComponent = creator.root

        const h1 = testComponent.findByType('h1');
        expect(h1.children[0]).toEqual('Current Movies ');

        const h3 = testComponent.findByType('h3');
        expect(h3.children[0]).toEqual('Come to our cinema and watch these films right now!');

        const tree = creator.toJSON();
        expect(tree).toMatchSnapshot();
    })

})
