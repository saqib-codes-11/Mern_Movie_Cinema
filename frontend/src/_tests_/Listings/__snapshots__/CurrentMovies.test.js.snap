// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current movies landing page testing should render forum landing page 1`] = `
<div
  class="background"
>
  <div
    className="container"
  >
    <div
      className="bgBlur jumbotron"
    >
      <h1
        className="landing-text"
      >
        Current Movies 
      </h1>
      <h3
        className="lead"
      >
        Come to our cinema and watch these films right now!
      </h3>
      <div>
        <div
          className="row"
        >
          <div
            className="p-2 col-lg-4 col-md-6 col-12"
          >
            <div
              className="card-wrapper"
            >
              <div
                className="content"
              >
                <div
                  className="face-front z-depth-2"
                >
                  <img
                    alt="Fast and Furious 9"
                    className="movieImage"
                    src="https://m.media-amazon.com/images/M/MV5BMjI0NmFkYzEtNzU2YS00NTg5LWIwYmMtNmQ1MTU0OGJjOTMxXkEyXkFqcGdeQXVyMjMxOTE0ODA@.jpg"
                  />
                </div>
                <div
                  className="face-back z-depth-2 colorScheme"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="card-title h5"
                    >
                      Fast and Furious 9
                    </div>
                    <div
                      className="card-title h5"
                    >
                      Directed by 
                      Justin Lin
                    </div>
                    <br />
                    <div
                      className="card-subtitle h6"
                    >
                      Released: 
                      2021
                    </div>
                    <a
                      href="/released/undefined"
                      onClick={[Function]}
                    >
                      <button
                        className="alignBottom btn btn-dark"
                        disabled={false}
                        type="button"
                      >
                        View more...
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
